src/lib/modules/job/job.go

```diff
@@ -0,0 +1,189 @@
+package job
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	config "github.com/bench-routes/bench-routes/src/lib/config_v2"
+	"github.com/bench-routes/bench-routes/src/lib/modules/evaluate"
+	"github.com/bench-routes/bench-routes/tsdb/file"
+)
+
+// Executable is an interface that is implemented by machineJob and monitoringJob
+type Executable interface {
+	Execute()
+	Abort()
+	Info() *JobInfo
+}
+
+type JobInfo struct {
+	mux         sync.RWMutex
+	Name        string
+	Every       time.Duration
+	lastExecute time.Time
+}
+
+// NewJob creates a new job based on the typ
+func NewJob(typ string, app file.Appendable, c chan struct{}, api *config.API) (Executable, error) {
+	switch typ {
+	case "machine":
+		job, err := newMachineJob(app, c, api)
+		if err != nil {
+			return nil, fmt.Errorf("error creating job : %w", err)
+		}
+		return job, nil
+	case "monitor":
+		job, err := newMonitoringJob(app, c, api)
+		if err != nil {
+			return nil, fmt.Errorf("error creating job : %w", err)
+		}
+		return job, nil
+	default:
+		return nil, fmt.Errorf("`typ` provided is invalid")
+	}
+}
+
+type monitoringJob struct {
+	JobInfo
+	app     file.Appendable
+	sigCh   chan struct{}
+	client  *http.Client
+	request *http.Request
+}
+
+func newMonitoringJob(app file.Appendable, c chan struct{}, api *config.API) (*monitoringJob, error) {
+	newJob := &monitoringJob{
+		app:   app,
+		sigCh: c,
+		JobInfo: JobInfo{
+			Name:        api.Name,
+			Every:       api.Every,
+			lastExecute: time.Now().Add(api.Every * -1),
+		},
+		client: &http.Client{
+			Timeout: time.Second * 5,
+		},
+		request: &http.Request{},
+	}
+	if err := newReq(newJob, api); err != nil {
+		return nil, fmt.Errorf("error making http request : %w", err)
+	}
+	return newJob, nil
+}
+
+// Execute executes the monitoringJob
+func (mn *monitoringJob) Execute() {
+	for range mn.sigCh {
+		mn.JobInfo.writeTime()
+		response, err := evaluate.Monitor(mn.client, mn.request)
+		if err != nil {
+			fmt.Println(fmt.Errorf("%s", err))
+		}
+		val := fmt.Sprintf("%v|%v", response.Delay.Microseconds(), strconv.Itoa(response.Length))
+		mn.app.Append(file.NewBlock("monitoring", val))
+	}
+}
+
+// Aborts aborts the running job
+func (mn *monitoringJob) Abort() {
+	close(mn.sigCh)
+}
+
+// Info returns jobInfo of the job
+func (mn *monitoringJob) Info() *JobInfo {
+	return &mn.JobInfo
+}
```

Comment:
This should be read-only, hence lets make it non-pointer.

```suggestion
func (mn *monitoringJob) Info() JobInfo {
	return mn.JobInfo
}
```